sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -746,6 +750,254 @@ case class DescribeTableCommand(
+
+      case structType: StructType =>
+        val fieldsJson = structType.fields
+          .map { field =>
+            val fieldTypeJson = jsonType(field.dataType)
+            s"""{"name": "${field.name}", "type": $fieldTypeJson, "nullable": ${field.nullable}}"""
+          }
+          .mkString(", ")
+        s"""{"type": "struct", "fields": [$fieldsJson]}"""
+
+      case decimalType: DecimalType =>
+        s"""{"type": "decimal", "precision": ${decimalType.precision},
+           |"scale": ${decimalType.scale}}""".stripMargin
+
+      case varcharType: VarcharType =>
+        s"""{"type": "varchar", "length": ${varcharType.length}}"""
+
+      case charType: CharType =>
+        s"""{"type": "char", "length": ${charType.length}}"""
+
+      // Only override TimestampType; TimestampType_NTZ type is already timestamp_ntz
+      case _: TimestampType =>
+        """{"type": "timestamp_ltz"}"""
+
+      case yearMonthIntervalType: YearMonthIntervalType =>
+        def getFieldName(field: Byte): String = {
+          YearMonthIntervalType.fieldToString(field)
+        }
+
+        val startUnit = getFieldName(yearMonthIntervalType.startField)
+        val endUnit = getFieldName(yearMonthIntervalType.endField)
+        s"""{"type": "interval", "start_unit": "$startUnit", "end_unit": "$endUnit"}"""
+
+      case dayTimeIntervalType: DayTimeIntervalType =>
+        def getFieldName(field: Byte): String = {
+          DayTimeIntervalType.fieldToString(field)
+        }
+
+        val startUnit = getFieldName(dayTimeIntervalType.startField)
+        val endUnit = getFieldName(dayTimeIntervalType.startField)
+        s"""{"type": "interval", "start_unit": "$startUnit", "end_unit": "$endUnit"}"""
+
+      // Base case for other simple types
+      case _ =>
+        s"""{"type": "${dataType.typeName}"}"""
+    }
+  }
+
+  private def normalizeStr(str: String): String = {
+    str.toLowerCase().replace(" ", "_")
+  }
+
+  private def describeColsJson(
+      schema: StructType,
+      jsonMap: mutable.LinkedHashMap[String, JValue],
+      header: Boolean): Unit = {
+
+    val defaultValuesMap = Option(ResolveDefaultColumns.getDescribeMetadata(schema))
+      .getOrElse(Seq.empty)
+      .collect { case (name, _, defaultValue) => name -> defaultValue }
+      .toMap
+
+    val columnsJson = schema.map { case column =>
+      val commentField = column.getComment()
+        .map(c => s""", "comment": "${c.replace("\"", "\\\"")}"""")
+        .getOrElse("")
+      val defaultValueJson = defaultValuesMap.get(column.name)
+        .map(defaultValue => s""", "default_value": "${defaultValue.replace("\"", "\\\"")}"""")
+        .getOrElse("")
+
+      s"""{
+         |  "name": "${column.name}",
+         |  "type": ${jsonType(column.dataType)}
+         |  $commentField$defaultValueJson
+         |}""".stripMargin
+    }.mkString("[", ",", "]")
+
+    jsonMap += "columns" -> parse(columnsJson)
```

Comment:
shall we generate JValue directly from the schema instead of generating JSON string and paring it here?